### DISCUSSION TITLE
Track origin of internal traffic from what part of page the person followed a link from

### DISCUSSION THREAD

**Discussion by mlncn** [plausible/analytics#1927] [Anything Else]
How to easily track internal origin of traffic (reached an article from trending vs editor's picks blocks, got to donate page via main menu vs footer links)?

The goal is to track things like: how many people clicked on links from what *parts* of pages?
 in the 'related content' block or 'popular content' block or the donate link and 

News sites seem to use `origin` for this.

(I think it would be really cool if this could be done simply by adding classes like `origin-footer` or `origin-main-menu` or `origin-highlighted` with a script extension adding the `?origin=footer` to all links within the region with the special class— but first we need a way to see the result of such query parameters at all.)

We would be happy to abuse `source` or even `ref` for this but it does not seem that would work?  This seems pretty fundamental, so wondering if i'm missing something.  Here is what we see in the filters:

  - UTM tags is only for UTM tags
  - ref and source are treated as aliases for UTM_SOURCE
  - filtering page by query (`?`) does not work (shows everything) and filtering by `=` shows only weird typos– https://plausible.io/plausible.io?period=month&page=**%3D**
  - given the above, even if `origin` and other were to be allowed by the standard script, or even if they were added with the manual script, there does not seem to be a good way to filter for them in the Plausible interface

Is there a good way forward?

**Comment by RobertJoonas** [comment]
Hi @mlncn!

> The goal is to track things like: how many people clicked on links from what parts of pages?

For that, you could just set up [custom events](https://plausible.io/docs/custom-event-goals) for these link clicks and send the different classes as custom properties.

> How to easily track the internal origin of traffic (reached an article from trending vs editor's picks blocks)

That sounds like a funnel for conversions (https://github.com/plausible/analytics/discussions/171). This would allow you to set up custom events for both the link click and the article pageview and then see the funnel for it. 

It doesn't work with UTM tags, because they are only used at the session level.  So basically if a visitor is already on your site before clicking the UTM-tagged link, then the UTM tags are ignored because the session is already active. UTM tags are used only when they **initiate** a new session on your site.

And you can't use page filters for UTM, as they're stripped from the URL. There's a [workaround in our docs](https://plausible.io/docs/custom-query-params) that you can use to make the UTM tags appear in your Pages reports. But yeah, even if you were able to filter by these tags, I'm not sure how that would help with this use case.
